data/transactions/logic/box.go

```diff
@@ -50,40 +50,44 @@ func (cx *EvalContext) availableBox(name string, operation int, createSize uint6
 	return nil
 }
 
-func createBox(cx *EvalContext, name string, value string, appAddr basics.Address) error {
+func createBox(cx *EvalContext, name string, value string, appAddr basics.Address) (bool, error) {
 	// Enforce length rules. Currently these are the same as enforced by
 	// ledger. If these were ever to change in proto, we would need to isolate
 	// changes to different program versions. (so a v7 app could not see a
 	// bigger box than expected, for example)
 	if len(name) == 0 {
-		return fmt.Errorf("box names may not be zero length")
+		return false, fmt.Errorf("box names may not be zero length")
 	}
 	if len(name) > cx.Proto.MaxAppKeyLen {
-		return fmt.Errorf("name too long: length was %d, maximum is %d", len(name), cx.Proto.MaxAppKeyLen)
+		return false, fmt.Errorf("name too long: length was %d, maximum is %d", len(name), cx.Proto.MaxAppKeyLen)
 	}
 	size := uint64(len(value))
 	if size > cx.Proto.MaxBoxSize {
-		return fmt.Errorf("box size too large: %d, maximum is %d", size, cx.Proto.MaxBoxSize)
+		return false, fmt.Errorf("box size too large: %d, maximum is %d", size, cx.Proto.MaxBoxSize)
 	}
 
 	err := cx.availableBox(name, boxCreate, size) // annotate size for write budget check
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	return cx.Ledger.NewBox(cx.appID, name, value, appAddr)
 }
 
 func opBoxCreate(cx *EvalContext) error {
-	last := len(cx.stack) - 1 // name
-	prev := last - 1          // size
+	last := len(cx.stack) - 1 // size
+	prev := last - 1          // name
 
-	name := string(cx.stack[last].Bytes)
-	size := cx.stack[prev].Uint
+	name := string(cx.stack[prev].Bytes)
+	size := cx.stack[last].Uint
 	appAddr := cx.getApplicationAddress(cx.appID)
 
-	cx.stack = cx.stack[:prev]
-	return createBox(cx, name, string(make([]byte, size)), appAddr)
+	cx.stack = cx.stack[:last]
+	existed, err := createBox(cx, name, string(make([]byte, size)), appAddr)
```

Comment:
I tried to follow the logic, and maybe I'm missing something. But my trace seems to indicate that `existed` has the opposite value of what I think it should have. In particular:
* For the vanilla case of brand new box:
  * `NewBox` returns `(true, nil)` &rarr;
    * `createBox` returns `(true, nil)`
      * therefore `existed == true`
* For the already existing box case:
  *  `NewBox` returns `(false, errOnMismatch(existing, value) )` &rarr;
    * `creatBox` returns `(false, errOnMismatch(existing, value) )`
      * therefore `existed == false`

Reply:
Yes, that's a dumb variable name from when I was looking at it the other way, but it's the proper behavior.  I'll change it to `created`.  I want `box_create` to be true if it actually does a create, just as `box_del` returns true is it actually performs a delete.